### PR TITLE
Simplify utility implementations in `spring-core`

### PR DESCRIPTION
**Pull request by polyglot-k**
You just received a pull request, so I changed the format similar to that.

- enhancement by adding early return
- delete the unnecessary 'else'

### CODE REVIEW

File: spring-core/src/main/java/org/springframework/util/FastByteArrayOutputStream.java [hunk 1]
```diff
@@ -504,13 +500,10 @@ public void updateMessageDigest(MessageDigest messageDigest) {
 		 */
 		@Override
 		public void updateMessageDigest(MessageDigest messageDigest, int len) {
-			if (this.currentBuffer == null) {
+			if (this.currentBuffer == null || len == 0) {
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/util/FastByteArrayOutputStream.java:503]: Please revert the changes to this method, since the subsequent inline comment does not apply to the `len == 0` case.

File: spring-core/src/main/java/org/springframework/util/LinkedCaseInsensitiveMap.java [hunk 2]
```diff
@@ -119,11 +119,10 @@ public boolean containsKey(Object key) {
 			}
 			@Override
 			protected boolean removeEldestEntry(Map.Entry<String, V> eldest) {
-				boolean doRemove = LinkedCaseInsensitiveMap.this.removeEldestEntry(eldest);
-				if (doRemove) {
+				if (LinkedCaseInsensitiveMap.this.removeEldestEntry(eldest)) {
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/util/LinkedCaseInsensitiveMap.java:122]: Please revert the changes to the `removeEldestEntry()` method, since the proposal changes the logic.

**Review by sbrannen — CHANGES_REQUESTED** [review]
I've requested that you revert a few changes.

**Comment by KNU-K** [comment]
Thank you for your review. I've finished working on the revers.

File: spring-core/src/main/java/org/springframework/util/FastByteArrayOutputStream.java [hunk 3]
```diff
@@ -507,8 +503,7 @@ public void updateMessageDigest(MessageDigest messageDigest, int len) {
 			if (this.currentBuffer == null) {
 				// This stream doesn't have any data in it...
 				return;
-			}
-			else if (len == 0) {
+			}else if(len == 0){
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/util/FastByteArrayOutputStream.java:506]: Please revert this unnecessary formatting change.

File: spring-core/src/main/java/org/springframework/util/MimeType.java [hunk 4]
```diff
@@ -583,7 +581,6 @@ public int compareTo(MimeType other) {
 				}
 			}
 		}
-
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/util/MimeType.java:586]: Please revert this unnecessary formatting change.

**Review by sbrannen — CHANGES_REQUESTED** [review]

**Comment by KNU-K** [comment]
@sbrannen 
I'm not used to contributing, but thank you for your meticulous review

File: spring-core/src/main/java/org/springframework/util/MimeType.java [hunk 5]
```diff
@@ -583,7 +581,7 @@ public int compareTo(MimeType other) {
...
+
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/util/MimeType.java:584]: I assume this will break the build, since we do not allow trailing whitespace.

In any case, please remove those tabs.

**Review by sbrannen — CHANGES_REQUESTED** [review]

File: spring-core/src/main/java/org/springframework/util/MimeType.java (hunk 5 above)
Comment (KNU-K) [reply spring-core/src/main/java/org/springframework/util/MimeType.java]: I just removed it

**Comment by sbrannen** [comment]
> When I checked the report because the build didn't pass, there was an error in the "reactive" part. However, I didn't touch that part, and it's not where I corrected it that depends.

Yep... `WebClientIntegrationTests.applyAttributesToNativeRequest()` is a flaky test that's been failing quite a lot lately, and the team is looking into how to address that.

So, you can safely ignore those failures in the interim.

**Comment by KNU-K** [comment]
Thank you for your good review of the long work :) 
Have a nice day today.

**Comment by sbrannen** [comment]
This has been merged into `6.2.x` and `main` in 2494ecb47bedb9c82c2a4c906005eb1cff8d5c22.

After merging, I noticed the copyright headers had not been updated, so I did that in 7f7819329c952a98a6e4ed81d376b5e8ca610e79.

Thanks for the PR and for making all of the requested changes.
